### PR TITLE
DM-43894: Avoid expected warnings in HSM higher order moments plugin

### PR DESCRIPTION
**Pull request by arunkannawadi**

### CODE REVIEW

File: python/lsst/meas/extensions/shapeHSM/_hsm_higher_moments.py
Comment (leeskelvin) [inline python/lsst/meas/extensions/shapeHSM/_hsm_higher_moments.py]: Does the record flag require the `centroid` parameter to be set? I'm not quite sure why you decided to move it above the raise here.

File: python/lsst/meas/extensions/shapeHSM/_hsm_higher_moments.py
Comment (leeskelvin) [inline python/lsst/meas/extensions/shapeHSM/_hsm_higher_moments.py]: Would an in-line comment above this `if` help explain the rationale for this new addition?

**Review by leeskelvin — APPROVED** [review]

File: python/lsst/meas/extensions/shapeHSM/_hsm_higher_moments.py
Comment (arunkannawadi) [reply python/lsst/meas/extensions/shapeHSM/_hsm_higher_moments.py]: Ah, nice catch, Lee! I meant to log (at the DEBUG level) that a measurement at this location failed. I realized that that gets logged anyway in the measurement framework here: https://github.com/lsst/meas_base/blob/main/python/lsst/meas/base/baseMeasurement.py#L407-L409

So I decided to skip logging here. I'll move the `centroid` back to where it was.

File: python/lsst/meas/extensions/shapeHSM/_hsm_higher_moments.py
Comment (arunkannawadi) [inline python/lsst/meas/extensions/shapeHSM/_hsm_higher_moments.py:372]: If the `HsmPsfMoments` algorithm failed because of lack of PSF, I'd like this plugin to also fail fatally instead of attempting every single source/object and failing individually. So I moved this up here.

**Review by arunkannawadi — COMMENTED** [review]
Lee, I changed more lines than I set out to. Would you mind taking a quick look again? I'll run Jenkins again before merging.

**Comment by leeskelvin** [comment]
Thanks Arun! I like the reformat here, which reads more cleanly to me than it was prior. I think the changes here make sense, and thank you for adding some in-line comments to help with that. I have nothing to add, LGTM!

PS - :crossed_fingers: for Jenkins

**Review by leeskelvin — APPROVED** [review]

**Pull request merged**
